crates/client/src/idl.rs

```diff
@@ -108,6 +114,76 @@ pub enum Error {
     MissingOrInvalidProgramItems(&'static str),
 }
 
+struct ModPub {
+    pub mod_name: String,
+    pub is_pub: bool,
+}
+
+struct FullPathFinder {
+    target_item_name: String,
+    current_module: String,
+    found_path: Option<String>,
+    module_pub: Vec<ModPub>,
+}
+
+fn find_item_path(code: &str, target_item_name: &str) -> Option<String> {
+    let syn_file = syn::parse_file(code).ok()?;
```

Comment:
I think it is not necessary to call the expensive parse_file function all over again every time the find_item_path is called. It would be better to pass it as a reference to the parsed file that is parsed at the beginning of the `parse_to_idl_program` function and therefore do the parsing only once...